app/repositories/proxy_types_repository.rb

```diff
@@ -0,0 +1,83 @@
+# frozen_string_literal: true
+
+class ProxyTypesRepository
+  attr_reader :base_url, :headers
+
+  def initialize(config:)
+    @config = config
+    @base_url = "#{rmapi_url}/rm/rmaccounts/#{config.customer_id}/proxies"
+
+    @headers = {
+      'X-Api-Key': config.api_key,
+      'Content-Type': 'application/json',
+      'Accept': 'application/json'
+    }
+  end
```

Comment:
so, now that there are multiple repositories, there are a bunch of things that we should pull out and centralize with respect to all repo classes.  i'm not saying we should do all of that here, but if there's time maybe we could do something easy like yank these headers out into a superclass?